File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length - 1];
        
        int min = Arrays.stream(arr).min().getAsInt();
        int j = 0;
        
        if(arr.length < 2) {
            return new int[] {-1};
        }
        for(int i = 0; i < arr.length; i++){
            if(arr[i] != min){
                answer[j] = arr[i];
                j++;
            }
            
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.go
func solution(arr []int) []int {
    if len(arr) == 1 {
        return []int{-1}
    }

    min := arr[0]
    for _, v := range arr {
        if v < min {
            min = v
        }
    }

    result := []int{}
    removed := false
    for _, v := range arr {
        if v == min && !removed {
            removed = true
            continue
        }
        result = append(result, v)
    }

    return result
}
